src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.sql

```diff
@@ -1629,6 +1702,59 @@ AS
     COMMIT TRANSACTION
 GO
 
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     AddExtendedQueryTagError
+--
+-- DESCRIPTION
+--    Adds an Extended Query Tag Error or Updates it if exists.
+--
+-- PARAMETERS
+--     @tagKey
+--         * The related extended query tag's key
+--     @errorCode
+--         * The error code
+--     @watermark
+--         * The watermark
+--     @createdTime
+--         * The time the error was created
+--
+-- RETURN VALUE
+--     The tag key of the error added.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.AddExtendedQueryTagError (
```

Comment:
Should'nt we also have a delete when tag or instance is deleted?

Reply:
I used to have a stored proc for delete, but I removed it since I was going to do the deletes in the deleteInstance stored procedure.

Reply:
We also need to delete the errors when a user possibly deletes the tag, right?